### PR TITLE
BYUC-193 create IAM Role for cross account access for kickstart and establish concourse pipeline

### PR DESCRIPTION
**Pull request by ashisprasad06**
Hello,

This pull request is to merge the following changes made as part of BYUC-193 below 

1. Concourse pipeline - This pipeline has been extended upto integration, pre-prod and production
2. Create new IAM Role - New IAM role is getting created using terraform. This role is required for cross account access to read data from source bucket. This IAM role needs to be deployed first before source can deploy their changes

### CODE REVIEW

File: aviator.yml [hunk 1]
```diff
@@ -6,7 +6,6 @@ spruce:
       - with_in: ci/
         regexp: ".*.yml"
       - with_in: ci/jobs/
-        regexp: ".*.yml"
```
Comment (dan-hill2802) [inline aviator.yml:9]: This should be kept. Why did you remove it? Was there an issue

Comment (ashisprasad06) [reply aviator.yml:9]: No issues, I just referred to one of the aviator.yml for aws-analytical-dataset-generation repo .. it didn't had that regex so thought to remove it .. if require I can keep it

File: src_connection_IAM.tf [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "dw_ksr_s3_readonly" {
+  name               = "dw_ksr_s3_readonly"
+  description        = "This is an IAM role which assumes role from UC side to gain temporary read access on S3 bucket"
+  assume_role_policy = data.aws_iam_policy_document.dw_ksr_assume_role.json
+  tags               = local.common_tags
+}
```
Comment (dan-hill2802) [inline src_connection_IAM.tf:19]: please add a new line to the end of the file

File: aviator.yml (hunk 1 above)
Comment (ashisprasad06) [reply aviator.yml]: Changes made in aviator.yml file to add regexp and to repo via commit id bf3244d079ed8129ccfe6eb220ccfc087b323408

File: src_connection_IAM.tf (hunk 2 above)
Comment (ashisprasad06) [reply src_connection_IAM.tf]: changes made in the src_connectivity_IAM.tf via commit id bf3244d079ed8129ccfe6eb220ccfc087b323408

File: src_connection_IAM.tf [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "dw_ksr_s3_readonly" {
+  name               = "dw_ksr_s3_readonly"
```
Comment (dan-hill2802) [inline src_connection_IAM.tf:15]: We really don't have a standard format for names, sadly. There's a mix of hyphens and under scores, but you don't need to state it's DataWorks, but I think the full name instead of ksr is better
```suggestion
  name               = "kick_start_s3_readonly"
```

Comment (ashisprasad06) [reply src_connection_IAM.tf:15]: can I put kickstart_s3_readonly, because this data is related to kickstart scheme launched by Govt., So it is just one word ?

Comment (dan-hill2802) [reply src_connection_IAM.tf:15]: yep, that would be fine

Comment (ashisprasad06) [reply src_connection_IAM.tf]: Changes has been made in the file the new role name is kickstart_s3_readonly

**Review by dan-hill2802 — APPROVED** [review]
